test/integration/JsonParsing.test.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  PrimitiveUri,
+  IElement,
+  IExtension,
+  Extension,
+  IKitchenSink,
+  KitchenSink,
+  PrimitiveTime,
+  Coding,
+  PrimitiveBoolean,
+  IResourceChild,
+  Resource,
+  ResourceChild, Bundle, BundleEntry,
+} from "../fixture/generatedTypeScript/fhir";
+import { IBundle } from "../fixture/generatedTypeScript/interfaces/IBundle";
+
+describe("Testing the Parsing logic of a generated class", () => {
+  function assertExtension(extensions?: Array<Extension>): void {
+    expect(extensions).toBeArrayOfSize(1);
+    const [extension] = extensions as Array<Extension>;
+    expect(PrimitiveBoolean.isPrimitiveBoolean(extension.value)).toBeTrue();
+    if (PrimitiveBoolean.isPrimitiveBoolean(extension.value)) {
+      expect(extension.value.value).toBeTrue();
+    }
+  }
+
+  it("should work around any circular dependencies", () => {
+    const element: IElement = { id: "val" };
+    expect(element.id).toBe("val");
+
+    const extension: IExtension = { valueString: "valString" };
+    expect(extension.valueString).toBe("valString");
+  });
+
+  it("should parse primitive JSON", () => {
+    const uri = PrimitiveUri.parsePrimitive("someVal");
+    expect(uri.value).toBe("someVal");
+  });
+
+  it("should parse primitive JSON with extensions", () => {
+    const extension: IElement = {
+      id: "extensionId",
+      extension: [
+        {
+          id: "idOfExtension",
+          url: "someUrl",
+          valueBoolean: true,
+        },
+      ],
+    };
+
+    const uri = PrimitiveUri.parsePrimitive("someVal", extension);
+    expect(uri.value).toBe("someVal");
+    expect(uri.id).toBe("extensionId");
+    expect(uri.extension).toBeDefined();
+    assertExtension(uri.extension);
+    const extensionArray = uri.extension as Array<Extension>;
+    const [extensionResult] = extensionArray;
+
+    expect(extensionResult.url?.value).toBe("someUrl");
+    expect(extensionResult.id).toBe("idOfExtension");
+    expect(extensionResult.value).toBeDefined();
+
+    expect(
+      PrimitiveBoolean.isPrimitiveBoolean(extensionResult.value)
+    ).toBeTrue();
+    if (PrimitiveBoolean.isPrimitiveBoolean(extensionResult.value)) {
+      expect(extensionResult.value.value).toBeTrue();
+    }
+  });
+
+  it("should parse complex FHIR JSON", () => {
+    const json: IKitchenSink = {
+      system: "123",
+      url: "http://something",
+      times: ["time1", "time2"],
+      singleCode: {
+        text: "someCode",
+      },
+      coding: [
+        {
+          version: "theVersion",
+        },
+      ],
+    };
+
+    const result = KitchenSink.parse(json);
+    expect(result).toBeDefined();
+    expect(result.system).toBe("123");
+    expect(result.url?.value).toBe("http://something");
+    expect(result.times).toBeArrayOfSize(2);
+    const [time1, time2] = result.times as Array<PrimitiveTime>;
+    expect(time1.value).toBe("time1");
+    expect(time2.value).toBe("time2");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5923420)